src/Compatibility/Microsoft.DotNet.ApiCompatibility/AssemblySymbolLoader.cs

```diff
@@ -260,7 +316,7 @@ public IEnumerable<IAssemblySymbol> LoadMatchingAssemblies(IEnumerable<IAssembly
                 if (warnOnMissingAssemblies && !found)
                 {
                     string assemblyInfo = validateMatchingIdentity ? assembly.Identity.GetDisplayName() : assembly.Name;
-                    _warnings.Add(string.Format(Resources.MatchingAssemblyNotFound, assemblyInfo));
+                    _warnings.Add(new AssemblyLoadWarning(DiagnosticIds.AssemblyNotFound, assemblyInfo, string.Format(Resources.MatchingAssemblyNotFound, assemblyInfo)));
```

Comment:
why are we storing these warnings ? wouldnt it better if we just log them

Reply:
Well the only reason that I see is that the AssemblySymbolLoader doesn't have a reference to the logger so, at least as of today, the way it works is that main function of the loader is to try to load the assemblies, and the runner which calls it, has the reference to the logger so it is able to gather the warnings and throw them. I don't fundamentally see a problem of passing the logger down to the loader as well, but I also don't see a problem with the way we have it working now, so unless there is a big reason why we would want to change it, I would keep it as is.